clients/src/main/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryUtils.java

```diff
@@ -204,16 +202,14 @@ public static byte[] compress(byte[] raw, CompressionType compressionType) throw
     public static ByteBuffer decompress(byte[] metrics, CompressionType compressionType) {
         ByteBuffer data = ByteBuffer.wrap(metrics);
         try (InputStream in = compressionType.wrapForInput(data, RecordBatch.CURRENT_MAGIC_VALUE, BufferSupplier.create());
-            ByteArrayOutputStream out = new ByteArrayOutputStream()) {
-
+             ByteBufferOutputStream out = new ByteBufferOutputStream(1)) {
```

Comment:
`1` is a little bit too small I think, to avoid too many expandBuffer invocation, maybe we can use 32 (which is the default byte array value in ByteArrayOutputStream) or 64 ? @chia7712  WDYT ?

Reply:
Yep, we can set a larger size to initialize buffer. Maybe we can use the same size (512 bytes) as line#192